src/Defaults/Trigger/User/UserEmailChanged.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * User email changed trigger
+ *
+ * @package notification
+ */
+
+declare(strict_types=1);
+
+namespace BracketSpace\Notification\Defaults\Trigger\User;
+
+use BracketSpace\Notification\Defaults\MergeTag;
+
+/**
+ * User email changed class
+ */
+class UserEmailChanged extends UserTrigger
+{
+	/**
+	 * User meta data
+	 *
+	 * @var array<mixed>|mixed
+	 */
+	public $userMeta;
+
+	/**
+	 * User new email address
+	 *
+	 * @var string
+	 */
+	public $userNewEmail;
+
+	/**
+	 * User new email address
+	 *
+	 * @var int|false
+	 */
+	public $userChangeEmailDatetime;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct()
+	{
+		parent::__construct(
+			'user/email_changed',
+			__('User email changed', 'notification')
+		);
+
+		$this->addAction(
+			'delete_user_meta',
+			10,
+			4
+		);
+
+		$this->setDescription(__('Fires when user changes his email address, after confirms by link.', 'notification'));
+	}
+
+	/**
+	 * Sets trigger's context
+	 *
+	 * @param array<int> $metaIds User meta Ids
+	 * @param int $objectId User ID
+	 * @param string $metaKey User meta key
+	 * @param string $metaValue User meta value
+	 * @return false|void
+	 */
+	public function context($metaIds, $objectId, $metaKey, $metaValue)
+	{
+		if ($metaKey !== '_new_email') {
+			return false;
+		}
+
+		/**
+		 * Make sure that the action comes from verifying the email address
+		 */
+		if (! isset($_REQUEST['newuseremail'])) {
+			return false;
+		}
+
+		$this->userId = $objectId;
+		$user = get_userdata($this->userId);
+
+		if (!$user instanceof \WP_User) {
+			return false;
+		}
+
+		$this->userObject = $user;
+		$this->userMeta = get_user_meta($this->userId);
+
+		$this->userRegisteredDatetime = strtotime($this->userObject->user_registered);
+		$this->userChangeEmailDatetime = time();
+
+		// phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+		$this->userNewEmail = $user->user_email;
+	}
+
+	/**
+	 * Registers attached merge tags
+	 *
+	 * @return void
+	 */
+	public function mergeTags()
+	{
+		parent::mergeTags();
+
+		$this->addMergeTag(new MergeTag\User\UserDisplayName());
+		$this->addMergeTag(new MergeTag\User\UserFirstName());
+		$this->addMergeTag(new MergeTag\User\UserLastName());
+		$this->addMergeTag(new MergeTag\User\UserNicename());
+
+		$this->addMergeTag(
+			new MergeTag\DateTime\DateTime(
+				[
+					'slug' => 'user_change_email_datetime',
+					'name' => __('User change email date time', 'notification'),
+				]
+			)
+		);
+
+		$this->addMergeTag(
+			new MergeTag\EmailTag(
+				[
+					'slug' => 'user_new_email',
```

Comment:
I don't think we'll need this in the end. It's exactly the same as the standard user email merge tag.
![image](https://github.com/BracketSpace/Notification/assets/18362490/21a6ddbe-d7c0-42ce-b8cd-2217316cdf87)